backend/src/bundles/notifications/notification.repository.ts

```diff
@@ -0,0 +1,101 @@
+import { type Repository } from '~/common/types/repository.type.js';
+
+import { NotificationEntity } from './notification.entity.js';
+import { type NotificationModel } from './notification.model.js';
+
+class NotificationRepository implements Repository {
+    private notificationModel: typeof NotificationModel;
+
+    public constructor(notificationModel: typeof NotificationModel) {
+        this.notificationModel = notificationModel;
+    }
+
+    public async find(
+        query: Record<string, unknown>,
+    ): Promise<NotificationEntity | null> {
+        const notification = await this.notificationModel
+            .query()
+            .findOne(query)
+            .execute();
+
+        return notification
+            ? NotificationEntity.initialize(notification)
+            : null;
+    }
+
+    public async findAll(userId: number): Promise<NotificationEntity[]> {
+        const notifications = await this.notificationModel
+            .query()
+            .where('userId', userId)
+            .orderBy('isRead', 'asc')
+            .orderBy('updated_at', 'desc')
+            .execute();
+
+        return notifications.map((notification) =>
+            NotificationEntity.initialize(notification),
+        );
+    }
+
+    public async paginatedFind({
+        userId,
+        page,
+        limit,
+    }: {
+        userId: number;
+        page: number;
+        limit: number;
+    }): Promise<{ items: NotificationEntity[]; total: number }> {
```

Comment:
can we add type helper for this?
something like 
type Paged<T> = { items: T, total: number }
Maybe you'll come up with a better name than that.